### PR TITLE
NOT MERGE: lets start unicode...

### LINKED ISSUE #2802: Encoding error while creating a package since conan 1.2.0+ with Python 2.7.14
Not sure how to reproduce, but I got a stacktrace when trying to package. Conan 1.0.4 and 1.1.1 works without errors. 
But with conan 1.2.0 and 1.2.3 I got the following output. (Conan installed with pip in Python 2.7.14)
```
Traceback (most recent call last):
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\command.py", line 1187, in run
    method(args[0][1:])
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\command.py", line 226, in create
    test_build_folder=args.test_build_folder)
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\conan_api.py", line 61, in wrapper
    return f(*args, **kwargs)
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\conan_api.py", line 323, in create
    keep_build=keep_build)
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\manager.py", line 395, in install
    installer.install(deps_graph, profile.build_requires, keep_build)
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\installer.py", line 264, in install
    root_conanfile)
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\installer.py", line 330, in _build
    keep_build, profile_build_requires, flat, deps_graph)
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\installer.py", line 385, in _build_package
    builder.package()
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\installer.py", line 134, in package
    install_folder, self._out)
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\packager.py", line 79, in create_package
    _create_aux_files(install_folder, package_folder, conanfile, copy_info)
  File "c:\python\2.7.14\lib\site-packages\conan-1.2.3-py2.7.egg\conans\client\packager.py", line 100, in _create_aux_files
    save(os.path.join(package_folder, CONAN_MANIFEST), str(digest))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xa0' in position 203792: ordinal not in range(128)

ERROR: 'ascii' codec can't encode character u'\xa0' in position 203792: ordinal not in range(128)

```
Currently I have no idea how to reproduce it with a small and simple package. But probably you have an idea or a fix by checking the stacktrace.

Best Aalmann

To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

### PR DESCRIPTION
**Pull request by memsharded**
Fix #2802

### CODE REVIEW

File: conans/unicode.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import six
+
+
+get_cwd = six.moves.getcwd
+
+
+def make_unicode(v):
+    if six.PY2:
+        if isinstance(v, str):
+            return v.decode("utf8")
```
Comment (lasote) [inline conans/unicode.py]: Just curious, how do you know that the str is utf8 and not other encoding like the evil windows one?

File: conans/client/manager.py [hunk 2]
```diff
@@ -137,6 +139,7 @@ def _load_install_conanfile(self, loader, reference_or_path):
             conanfile = loader.load_virtual([reference_or_path])
         else:
             output = ScopedOutput("PROJECT", self._user_io.out)
+            reference_or_path = make_unicode(reference_or_path)
```
Comment (lasote) [inline conans/client/manager.py]: Should be converted sooner more close to the command as soon as we know it is a path? It feels a little spread.

Comment (memsharded) [reply conans/client/manager.py]: Yes, I tried so. But then it happens that you have such conversion in 10x more places, I tried to be minimal

File: conans/unicode.py (hunk 1 above)
Comment (memsharded) [reply conans/unicode.py]: Good question. I am trying to understand...
```bash
>>> sys.getfilesystemencoding()
'mbcs'
>>> sys.getdefaultencoding()
'ascii'
>>> sys.stdout.encoding
'cp850'
```

**Comment by memsharded** [comment]
This is not very clear, can be problematic.
Also, the path for deprecating python 2.7 has already started, so this might be solved when using python 3 only. Moving it to conan 2.0.

**Comment by memsharded** [comment]
Closing this, very outdated, and it probably make sense to wait until moving to Python 3 only (end of this year, likely).

**Pull request closed without merging**